backend/apis/leads/get-lead.api.js

```diff
@@ -0,0 +1,144 @@
+const {
+  app,
+  databaseError,
+  pool,
+  invalidRequest,
+  notFound
+} = require("../../server");
+const mysql = require("mysql");
+const { checkValid, validId } = require("../utils/validation-utils");
+const {
+  responseFullName,
+  responseBoolean,
+  responseDateWithoutTime
+} = require("../utils/transform-utils");
+
+app.get("/api/leads/:id", (req, res, next) => {
+  const validationErrors = checkValid(req.params, validId("id"));
+
+  if (validationErrors.length > 0) {
+    return invalidRequest(res, validationErrors);
+  }
+
+  getLeadById(req.params.id, (err, lead) => {
+    if (err) {
+      return databaseError(req, res, err);
+    }
+
+    if (lead) {
+      res.send({
+        lead
+      });
+    } else {
+      notFound(res, `Could not find lead with id ${req.params.id}`);
+    }
+  });
+});
+
+exports.getLeadById = getLeadById;
+
+function getLeadById(leadId, cbk, connection) {
+  leadId = Number(leadId);
+
+  const getLead = mysql.format(
+    `
+      SELECT
+        leads.id AS leadId,
+        leads.dateOfSignUp,
+        leads.firstContactAttempt,
+        leads.secondContactAttempt,
+        leads.thirdContactAttempt,
+        leads.inactivityReason,
+        leads.firstName,
+        leads.lastName,
+        leads.phone,
+        leads.smsConsent,
+        leads.zip,
+        leads.age,
+        leads.gender,
+        leads.clientId,
+        leads.isDeleted,
+        leads.dateAdded,
+        leads.dateModified,
+        leads.addedBy,
+        leads.modifiedBy,
+        created.firstName AS createdByFirstName,
+        created.lastName AS createdByLastName,
+        modified.firstName AS modifiedByFirstName,
+        modified.lastName AS modifiedByLastName,
+        JSON_ARRAYAGG(
```

Comment:
I've never used JSON_ARRAYAGG before - looks like a nice solution for aggregating all the leads into one row.